### PR TITLE
issue: 3925885 Adjusting XLIO default allocation logic

### PR DESCRIPTION
**Pull request by BasharRadya**
- Modified the default ring allocation logic from ‘ring per interface’ to ‘ring per thread’.
- Adjusted the MCE_SPEC_NGINX_DPU and MCE_SPEC_NGINX by setting ring_allocation_logic_tx and ring_allocation_logic_rx to ‘ring per interface’.
- Fixed a minor side-effect in the ring_allocation_logic constructor.

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

File: src/core/dev/net_device_val.cpp [hunk 1]
```diff
@@ -70,8 +70,12 @@
 #define nd_logfunc    __log_info_func
 #define nd_logfuncall __log_info_funcall
 
+//This constructor is called to initialize class members of type ring_alloc_logic_attr that are not
```
Comment (AlexanderGrissik) [inline src/core/dev/net_device_val.cpp:73]: The first 3 lines are obvious and not needed.
Its enough to say that the value given in this constructor is later always overridden.

**Comment by galnoam** [comment]
@BasharRadya , please fix conflicts.
Thanks

**Comment by pasis** [comment]
Should we disable progress engine by default as well? @AlexanderGrissik, WDYT?

**Comment by BasharRadya** [comment]
bot:retest

**Comment by galnoam** [comment]
@dpressle, can you check why Bashars PRs do not trigger CI?

**Comment by dpressle** [comment]
bot:retest

**Comment by BasharRadya** [comment]
bot:retest

**Comment by galnoam** [comment]
@dpressle, we have issue with the CI for some reason for Bashar commits, can you look?

**Comment by BasharRadya** [comment]
bot:retest

**Comment by galnoam** [comment]
@BasharRadya, please check why tests are failing.
regarding blackduck fail, its known issue.

**Comment by AlexanderGrissik** [comment]
bot:retest

**Review by AlexanderGrissik — APPROVED** [review]

**Comment by BasharRadya** [comment]
bot:retest

**Comment by iftahl** [comment]
bot:retest

**Comment by BasharRadya** [comment]
bot:retest

**Comment by BasharRadya** [comment]
bot:retest

**Comment by BasharRadya** [comment]
bot:retest

**Comment by BasharRadya** [comment]
bot:retest

**Pull request merged**